code/iaas/api-logic/src/main/java/io/cattle/platform/iaas/api/auth/config/GithubConfigManager.java

```diff
@@ -0,0 +1,208 @@
+package io.cattle.platform.iaas.api.auth.config;
+
+import io.cattle.platform.archaius.util.ArchaiusUtil;
+import io.cattle.platform.core.model.Setting;
+import io.cattle.platform.deferred.util.DeferredUtils;
+import io.cattle.platform.iaas.api.auth.github.GithubAccountInfo;
+import io.cattle.platform.iaas.api.auth.github.GithubClient;
+import io.cattle.platform.json.JsonMapper;
+import io.cattle.platform.object.ObjectManager;
+import io.github.ibuildthecloud.gdapi.exception.ClientVisibleException;
+import io.github.ibuildthecloud.gdapi.factory.SchemaFactory;
+import io.github.ibuildthecloud.gdapi.model.ListOptions;
+import io.github.ibuildthecloud.gdapi.request.ApiRequest;
+import io.github.ibuildthecloud.gdapi.request.resource.impl.AbstractNoOpResourceManager;
+import io.github.ibuildthecloud.gdapi.util.ResponseCodes;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import javax.inject.Inject;
+
+import org.apache.commons.lang3.StringUtils;
+
+import com.google.common.base.Function;
+import com.google.common.collect.Lists;
+import com.netflix.config.DynamicBooleanProperty;
+import com.netflix.config.DynamicStringProperty;
+
+public class GithubConfigManager extends AbstractNoOpResourceManager {
+
+    private static final String GITHUB_CONFIG = "githubconfig";
+
+    private static final DynamicBooleanProperty SECURITY = ArchaiusUtil.getBoolean("api.security.enabled");
+    private static final DynamicStringProperty GITHUB_CLIENT_ID = ArchaiusUtil.getString("api.auth.github.client.id");
+    private static final DynamicStringProperty GITHUB_ALLOWED_USERS = ArchaiusUtil.getString("api.auth.github.allow.users");
+    private static final DynamicStringProperty GITHUB_ALLOWED_ORGS = ArchaiusUtil.getString("api.auth.github.allow.orgs");
+
+    private JsonMapper jsonMapper;
+    private ObjectManager objectManager;
+    private GithubClient client;
+
+    @Override
+    public Class<?>[] getTypeClasses() {
+        return new Class<?>[] { GithubConfig.class };
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    protected Object createInternal(String type, ApiRequest request) {
+        if (!StringUtils.equals(GITHUB_CONFIG, request.getType())) {
+            return null;
+        }
+        Map<String, Object> config = jsonMapper.convertValue(request.getRequestBodyObject(), Map.class);
```

Comment:
Argh.... Don't use RequestBodyObject().  I have to change the API to make this not possible.  That call is unsafe.